library/core/src/panicking.rs

```diff
@@ -56,6 +56,16 @@ pub const fn panic_str(expr: &str) -> ! {
     panic_display(&expr);
 }
 
+#[cfg(not(bootstrap))]
+#[inline]
+#[track_caller]
+#[lang = "unreachable_display"] // needed for `non-fmt-panics` lint
```

Comment:
If a lang item is only used for diagnostics, it's better to use `rustc_diagnostic_item` instead.

Reply:
(I see that `panic_str` has the same comment. Looks like that's left over from this cleanup: #90273. Originally it wasn't only there for diagnostics. I'll clean up `panic_str` some time soon.)